### DISCUSSION TITLE
Unable to make UICalendarSelectionSingleDateDelegate working on iOS 17.5 in my handler

### DISCUSSION THREAD

**Discussion by christophe-mommer** [dotnet/maui#26381] [Q&A]
### Description

I wrote a custom handler as a proof of concept to demonstrate the usage in iOS. I did the same for Android. No issue on Android side, but on iOS, I'm unable to make the selection fire.
You can find the source code to demonstrate the issue here : https://github.com/christophe-mommer/MAUICalendarHandlerBug

@brminnick this is the same code as your course published at Dometrain, but you didn't test the selection in the videos (and you forgot to link the selectiondelegate to the native component)

### Steps to Reproduce

Put a breakpoint in "DidSelectDate" of the custom delegate, never fired when selecting a date in the app

### Version with bug

8.0.71 SR7.1

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

**Comment by ninachen03** [comment]
I can repro this issue at iOS platform on the latest 17.11.0 Preview 7.0( 8.0.80 & 8.0.71 & 8.0.61).


![image](https://github.com/user-attachments/assets/896fa9da-3737-47f2-8286-e4537a52a879)

**Comment by mattleibow** [comment]
Does this code work in a plain .NET for iOS app? I have not used this API, but it all looks like the code chat GPT wrote, so I am wondering if there is some other thing that is making it not work. Testing outside of MAUI may help.

Are there any errors or warnings from iOS in the debug console?

@brminnick do you know if selection worked in your tests?

**Reply by mattleibow** [reply]
@christophe-mommer I think this may not be a maui issue but an implementation issue. If it turns out to be an actual maui issue, then we can move it back to issues. But for now I think a discussion with folks on getting this working in an ios app first is the first step.

**Reply by TheCodeTraveler** [reply]
Hey Matt! I'll be updating my code to .NET and re-publishing it to DomeTrain in the next few weeks. 

I'll double check this when I'm digging into it 👍

**Reply by TheCodeTraveler** [reply]
Hey @mattleibow! I confirmed that it was just a bug in my code. I can confirm that there is no issue in .NET iOS for `UICalendarSelectionSingleDateDelegate`.

Apologies @christophe-mommer! Long story short, I messed up two things in the video course:
- My `CalendarSelectionSingleDateDelegate` inherited from `IUICalendarSelectionSingleDateDelegate` instead of `UICalendarSelectionSingleDateDelegate`
- I failed to assign `UICalendarView.SelectionBehavior`

I've open-sourced the code and you can find it here: https://github.com/Dometrain/deep-dive-net-maui

**Reply by christophe-mommer** [reply]
@TheCodeTraveler I did exactly what you metionned in the code and it appears that the calendar remains in a readonly mode, selection cannot be done.

With this code 

```
public partial class CustomCalendarHandler : ViewHandler<ICalendar, UICalendarView>, IDisposable
{
    private UICalendarSelection? _selectionDelegate;
    protected override UICalendarView CreatePlatformView()
    {
        _selectionDelegate = new UICalendarSelectionSingleDate(new CalendarSelectionSingleDateDelegate(VirtualView));
        return new UICalendarView
        {
            Calendar = new(NSCalendarType.Gregorian),
            SelectionBehavior = _selectionDelegate,

        };
    }

    protected override void ConnectHandler(UICalendarView platformView)
    {
        base.ConnectHandler(platformView);

    }

    protected override void DisconnectHandler(UICalendarView platformView)
    {
        base.DisconnectHandler(platformView);
        _selectionDelegate = new UICalendarSelectionSingleDate(new CalendarSelectionSingleDateDelegate(VirtualView));
    }

    static void MapSelectedDate(CustomCalendarHandler handler, ICalendar virtualView)
    {
        if (handler._selectionDelegate is UICalendarSelectionSingleDate calendarSelection)
        {
            if (virtualView.SelectedDate is null)
            {
                calendarSelection.SetSelectedDate(null, true);
                return;
            }
            calendarSelection.SetSelectedDate(new NSDateComponents
            {
                Year = virtualView.SelectedDate.Value.Year,
                Month = virtualView.SelectedDate.Value.Month,
                Day = virtualView.SelectedDate.Value.Day
            }, true);
        }
    }

    public void Dispose()
    {
        _selectionDelegate?.Dispose();
    }

    sealed class CalendarSelectionSingleDateDelegate(ICalendar calendar) : UICalendarSelectionSingleDateDelegate
    {
        public override bool CanSelectDate(UICalendarSelectionSingleDate selection, NSDateComponents? dateComponents)
            => true;

        public override void DidSelectDate(UICalendarSelectionSingleDate selection, NSDateComponents? date)
        {
            selection.SelectedDate = date;
            calendar.SelectedDate = date?.Date.ToDateTime();
            calendar.OnSelectedDateChanged(calendar.SelectedDate);
        }
    }
}
```
